_posts/2015-03-stitch-fix-heart-unix.md

```diff
@@ -0,0 +1,670 @@
+  2  [|                                                  0.6%]     Load average: 1.72 1.58 1.51
+  3  [||||||||                                          13.2%]     Uptime: 2 days, 03:32:02
+  4  [|                                                  0.7%]
+  5  [||||                                               5.2%]
+  6  [                                                   0.0%]
+  7  [|||||                                              6.5%]
+  8  [                                                   0.0%]
+  Mem[|||||||||||||||||||||||||||||||||||        8369/16384MB]
+  Swp[                                               0/1024MB]
+
+  PID USER     PRI  NI  VIRT   RES   SHR S CPU% MEM%   TIME+  Command
+75871 simeon    31   0 2407M  2144     0 C  0.0  0.0  0:00.00 htop
+
+```
+
+<a name="easily-manage-python-virtual-environments"></a>
+### Easily Manage Python Virtual Environments
+
+&#8212; [Jeff Magnusson](https://www.linkedin.com/in/jmagnuss) *(OS X / bash)*
+
+#### Problem
+Managing multiple Python virtual environments is tedious.
+
+#### Solution
+Install `pyenv` to your `~/.bashrc`.
+
+#### Script
+```bash
+PYTHON_VIRTUALENV_BASEPATH="$HOME/python/virtualenv"
+
+function pyenv {
+    if [ -z "$1" ]; then
+        echo `ls $PYTHON_VIRTUALENV_BASEPATH`
+    elif [ $1 == '--create' ]; then
+        pushd $PYTHON_VIRTUALENV_BASEPATH; virtualenv --no-site-packages $2; popd;
+    else
+        source $PYTHON_VIRTUALENV_BASEPATH/$1/bin/activate;
+    fi
+}
+```
+
+`pyenv` is a light wrapper around Python's `virtualenv` command.  Executed with no arguments (`pyenv`), it returns a list of currently installed virtual environments.  
+
+Executed with a single argument, it attempts to activate the `virtualenv` passed as the argument (`pyenv my_virtual_env`).  
+
+Executed with `--create` flag, it creates the virtual environment passed in the second argument (`pyenv --create my_virtual_env`).
+
+<a name="productivity"></a>
+## Productivity
+
+<a name="alias-everything"></a>
+### Alias Everything
+&#8212; [Eric Gravert](https://twitter.com/egravert) *(OS X, Linux / bash)*
+
+#### Problem
+I hate typing. 
+
+#### Solution
+I alias EVERYTHING.
+
+##### Command
+```
+alias be="bundle exec"
+alias brake="be rake"
+alias bspec="be rspec"
+alias clock='date "+DATE: %Y-%m-%d%nTIME: %r"'
+```
+
+##### Problem
+Typing an alias is still too many keystrokes.
+
+##### Solution
+Create custom key bindings.
+
+##### Command
+```
+bind -x '"\C-t"':clock
+```
+
+##### Input
+```
+^t
+```
+
+##### Output
+```
+DATE: 2015-03-06
+TIME: 01:08:53 PM
+```
+
+<a name="tab-complete-cd"></a>
+### Tab Completion
+&#8212; [Eric Gravert](https://twitter.com/egravert) *(OS X, Linux / bash)*
+
+#### Problem
+I still hate typing.
+
+#### Solution
+You can use `CDPATH` for tab completion of the `cd` command.  If I have a directory with all of my projects, I can `export` `CDPATH` to this directory.
```

Comment:
"You can set the `CDPATH` variable to add tab completion to the `cd` bash command. For example, if you have a directory which holds your projects, you can add that directory to the CDPATH variable to get tab completion of the project directories from anywhere in your file system.